numpy/random/meson.build

```diff
@@ -44,10 +44,7 @@ _bounded_integers_pyx = custom_target('_bounded_integer_pyx',
   command: [tempita_cli, '@INPUT@', '-o', '@OUTPUT@'],
 )
 
-c_args_random = [
-  cflags_large_file_support,
-  '-DNPY_NO_DEPRECATED_API=0',  # Cython still uses old NumPy C API
```

Comment:
OK, so we need to actually set a value here, otherwise the following warning triggers
https://github.com/numpy/numpy/blob/6fbd06bf3309aae73aa2bfacece1d91c4800fc07/numpy/_core/include/numpy/npy_1_7_deprecated_api.h#L9

However, the _right_ value to set this to would be the `NPY_API_VERSION`, which is only defined in `numpy/_core`; is it possible to "import" this value here somehow?

The bigger hammer would of course be to just delete that warning (or ever more things related to `NPY_NO_DEPRECATED_API`), since the `Cython>=3` lower bound means we don't need to use the deprecated API anywhere AFAIU

Reply:
Looks like it makes more sense to remove this warning(NPY_NO_DEPRECATED_API). Will lead to more deletions but it will be for better.

Reply:
You're both misunderstanding - the warning is needed if we ever want to turn that API usage into an error. You can't just no longer emit a deprecation warning. Removing the warning is going in the wrong direction unless we decide we never want to enforce the deprecation (if so, this PR isn't the place for it).

> However, the _right_ value to set this to would be the `NPY_API_VERSION`, which is only defined in `numpy/_core`; is it possible to "import" this value here somehow?

Also not quite. See https://github.com/scipy/scipy/blob/c0065b69a98be549b46e44573bbf3c13e98681da/scipy/meson.build#L75-L78

Reply:
I don't understand the size of the PR: it now should be perfectly fine to just use `=NPY_API_VERSION` here and call it a day?

Reply:
Yeah, this was me just hacking around to try to fix #27314 and misdiagnosing things in the process (which is why this PR is a draft). If there's something to salvage from this idea (e.g. some clean-up now that we require cython>=3), please let me know, otherwise let's close this PR.

Reply:
The only thing is fixing that one line.  Which I guess could be this PR (removing everything else).  I don't really see a purpose in the rest.